resources/world/world_data.gd

```diff
@@ -402,6 +442,13 @@ func get_cell_type(cell_index : int) -> int:
 
 func set_cell_type(cell_index : int, value : int):
 	if cell_index >= 0:
+		if cell_type[cell_index] == CellType.STARTING_ROOM and value == CellType.ROOM:
```

Comment:
This introduces extremely surprising behavior, and should be avoided. Trying to set a cell to a specific value should always set it to that value, unless it's impossible for very strong reasons, in which case, the value should stay the same, but the lack of change should be communicated to the caller somehow (If a situation like that ever appears, this function should be modified to return some error value, or at least a boolean to indicate success)

Since you need to know this information for other things to work, here are a few alternatives:
- Use cell meta, and store the information that the cell is part of the starting room in there (not that good, will need a lot of cells with this data, and modifying the starting room requires modifying all the meta data)

- Make a new function that checks if a cell is within a given room type, or returns the room type of a given cell. This would preferably be done by checking the cell against the rooms Rect2s or Polygons (polygons not implemented yet), instead of checking if the cell is in the cell array. This way, you could check if the cell is in the starting cell when you need this information.

- Make a new dense array (something like a PoolIntArray) that stores, for every cell, the room type they belong to. this would be similar to the cell_type array, but allow for room specific information, where cell_type would be a more generic thing.

Reply:
I don't think this is needed anymore since we have the RoomData, but I have to refactor STARTING_ROOM to stop existing and use `rooms["starting_room"]`, but that's outside of the scope of this PR because I'd have to refactor the previous two PRs. Already talked about it with @Mooses2k 

before doing this I checked and this was the only override I wanted to block, so pretty safe for now until I work on the starting room refactor for making cultist spawn and initial loot spawn follow GenerationSteps next week.